scripts/check_ecosystem.py

```diff
@@ -294,9 +295,47 @@ async def main(*, ruff1: Path, ruff2: Path, projects_jsonl: Optional[Path]) -> N
                 print()
                 print("</p>")
                 print("</details>")
+
+                # Count rule changes
+                for line in diff_str.splitlines():
+                    # Find rule change for current line or construction
+                    # + <rule>/<path>:<line>:<column>: <rule_code> <message>
+                    matches = re.findall(r": [A-Z]{1,3}[0-9]{3,4}", line)
+                    if len(matches) == 0:
+                        # Handle case where there are no regex matches e.g.
+                        # +                 "?application=AIRFLOW&authenticator=TEST_AUTH&role=TEST_ROLE&warehouse=TEST_WAREHOUSE" # noqa: E501, ERA001
+                        # Which was found in local testing
+                        continue
+
+                    rule_code = matches[0][2:]  # Trim leading ": "
```

Comment:
i'd use a [match group](https://docs.python.org/3/howto/regex.html#grouping) here

Reply:
sorry, i wasn't precise enough: my idea was to put the rule into a group and then just use the group so you don't need to manually trim

Reply:
Something more along the lines of 

```python
                for line in diff_str.splitlines():
                    # Find rule change for current line or construction
                    # + <rule>/<path>:<line>:<column>: <rule_code> <message>
                    matches = re.search(r": ([A-Z]{1,3}[0-9]{3,4})", line)

                    if matches is None:
                        # Handle case where there are no regex matches e.g.
                        # +                 "?application=AIRFLOW&authenticator=TEST_AUTH&role=TEST_ROLE&warehouse=TEST_WAREHOUSE" # noqa: E501, ERA001
                        # Which was found in local testing
                        continue

                    rule_code = matches.group(1)
```

I have pushed this in a commit, but due to the current [GitHub issues](https://www.githubstatus.com/), it does not appear to be showing up.

Reply:
Updated in f9e09c5. I'm not sure why this is not showing up in this PR

Reply:
yeah yesterday CI has also been flaky all day :/

thanks for fixing!